src/inboxsdk.d.ts

```diff
@@ -321,16 +321,36 @@ export class RouteView extends EventEmitter {
 export interface RowDescriptor {
   /** First textual column */
   title: string;
-  /** Second textual column */
-  body: string;
+  /** Second textual column. After {@link RowDescriptor#labels} if they're provided. */
+  body: string | ((args: { el: HTMLElement; onDestroy(): void }) => void);
+  /**
+   * Render an HTMLElement in the attachment icon area. This is often used to render an icon for the attachment type.
+   *
+   * @TODO do we need some sort of destroy hook here?
+   */
+  renderAttachmentIcon?(args: { el: HTMLElement; onDestroy(): void }): void;
   /** Last text right-aligned. Often used for dates. */
   shortDetailText: string;
   /**
    * Whether the row should be rendered as read or unread similar to Gmail styles.
+   * Defaults to unread which is 700 weight text and, in a light Gmail theme, a white background.
    *
    * @TODO is this actually required like the docs say?
    */
-  isRead?: string;
+  isRead?:
```

Comment:
yeah this starting to feel super weird as an API. 

I'd suggest the following:
- recipients, subject, snippet and detail properties can all take in Strings or an HtmlElement (or html string if we decide on that).
- isRead is just a boolean
- isRead controls the background style as well as the styles of recipients, subject, snippet and detail properties. But if the user passed in an HTML Element for any of those properties then they obv dont get styled
- on the Streak side we'd pass an HTMLElement for all those properties so we can get the text styling we want and we'd also pass isRead = false

Reply:
Chatted about this offline. We'd keeping the falsey value for `isRead`. 

In addition to that, `RowDescriptor#title` and `RowDescriptor#shortDetailText`, the Gmail thread row recipients and dates' equivalent, would also support an arbitrary HTMLElement option.

Reply:
If we supported the app passing an HTML element for the whole row, and the app could just set its own background color and font weight directly for everything, then it would make a lot of sense to keep this property really basic as @aleemstreak you're suggesting and not split out the parts of it.
But with the SDK in control of the row and the app just slotting in parts (which is a design we want so the SDK can handle aligning the columns based on the current Gmail settings etc), the app wouldn't have any direct way to control the background color of the full row and it would a weird API if switching any of the parts from a string to an HTMLElement forcibly switched the background color for the whole row.

I'm in favor of the API by @wegry here. I've put some commit suggestions here to make the description much more clear, and to suggest that the default behavior is consistent with what we're doing for Streak.

Reply:
alright fair

Reply:
I've included @Macil's suggestion blocks in [a7123b6](https://github.com/InboxSDK/InboxSDK/pull/1133/commits/a7123b66a88934643f2c84c7c780bc48f60686ee) through [7c13eeb](https://github.com/InboxSDK/InboxSDK/pull/1133/commits/7c13eebce28d0feaef37d3df8618da0249496516).